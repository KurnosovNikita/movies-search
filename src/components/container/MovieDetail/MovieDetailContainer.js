import React, {Fragment} from "react";
import HeaderDetailContainer from "../HeaderDetail/HeaderDetailContainer";
import CardsContainer from "../CardsContainer/CardsContainer";
import TextBlock from "../../presentational/TextBlock";
import {SORT_ORDER, SORT_TYPE, SEARCH_BY, getMovies} from '../../../utils';

class AppContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            movies: [],
        };
    }

    render() {
        const {movies} = this.state;
        const {selectedMovie, changeToGlobalSearchPage} = this.props;

        return (
          <Fragment>
              <HeaderDetailContainer
                selectedMovie={selectedMovie}
                changeToGlobalSearchPage={changeToGlobalSearchPage}/>
              <main className="main">
                  <TextBlock className="sort-by-genre" text={`Films by ${selectedMovie[0].genres[0]} genre`}/>
                  <CardsContainer movies={movies}
                                  handleImageClick={this.handleImageClick}/>
              </main>
          </Fragment>
        );
    }

    handleImageClick = (event) => {
        let selectedMovie = this.state.movies.filter(
          (movie) => movie.id === Number(event.target.parentNode.getAttribute('data-movie-id')));
        if (selectedMovie)
            this.props.changeToMovieDetailPage({selectedMovie});
    };

    componentWillMount() {
        this.fetchMovies();
    }

    fetchMovies = () => {
        getMovies({
            sortBy: SORT_TYPE.rating,
            sortOrder: SORT_ORDER.descending,
            search: this.props.selectedMovie[0].genres[0],
            searchBy: SEARCH_BY.genres,
            success: (resp) => {
                this.setState({movies: resp.data});
            },
        });
    };

    shouldComponentUpdate(nextProps, nextState) {
        return !(this.state.movies.length === nextState.movies.length
          && this.props.selectedMovie[0].id === nextProps.selectedMovie[0].id);
    }
}

export default AppContainer;