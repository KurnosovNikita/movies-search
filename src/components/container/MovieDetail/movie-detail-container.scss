.main {

    .sort-by-genre {
        font-size: 14px;
        font-weight: bold;
        margin: 10px 0 0 60px;
    }
}