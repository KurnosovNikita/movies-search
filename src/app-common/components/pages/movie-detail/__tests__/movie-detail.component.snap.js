import React from 'react';
import {shallow} from 'enzyme';
import MovieDetail from '../movie-detail.component';
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({});

describe('MovieDetail component', () => {
    let props,
      state;
    beforeEach(() => {


        state = {
            movies: [],
        };
        props = {
            selectedMovie: [{
                id: 505,
                genres: ['drama', 'comedy', 'triller'],
                title: 'Harry Potter',
                release_date: '20.05.2018',
                poster_path: 'https://puu.sh/Dl8Lx/d4fcc4e5e6.png',
            }],
            changeToGlobalSearchPage: jest.fn(),
        };

    });

    it('should render <MovieDetail> correctly', () => {
        const component = shallow(
          <Provider store={store}>
              <MovieDetail {...props}/>
          </Provider>);

        expect(component).toMatchSnapshot();
    });
});
