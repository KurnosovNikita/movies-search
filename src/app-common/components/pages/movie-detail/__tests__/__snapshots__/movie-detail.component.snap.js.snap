// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail component should render <MovieDetail> correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <ConnectFunction
    changeToGlobalSearchPage={[MockFunction]}
    selectedMovie={
      Array [
        Object {
          "genres": Array [
            "drama",
            "comedy",
            "triller",
          ],
          "id": 505,
          "poster_path": "https://puu.sh/Dl8Lx/d4fcc4e5e6.png",
          "release_date": "20.05.2018",
          "title": "Harry Potter",
        },
      ]
    }
  />
</ContextProvider>
`;
